monitors/test/ContractMonitor.js

```diff
@@ -0,0 +1,295 @@
+      expirationTimestamp: expirationTime.toString(),
+      withdrawalLiveness: "1000",
+      collateralAddress: collateralToken.address,
+      finderAddress: Finder.address,
+      tokenFactoryAddress: TokenFactory.address,
+      timerAddress: Timer.address,
+      priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"),
+      syntheticName: "Test UMA Token",
+      syntheticSymbol: "UMATEST",
+      liquidationLiveness: "10",
+      collateralRequirement: { rawValue: toWei("1.5") },
+      disputeBondPct: { rawValue: toWei("0.1") },
+      sponsorDisputeRewardPct: { rawValue: toWei("0.1") },
+      disputerDisputeRewardPct: { rawValue: toWei("0.1") },
+      minSponsorTokens: { rawValue: toWei("1") }
+    };
+
+    // Create a sinon spy and give it to the SpyTransport as the winston logger. Use this to check all winston
+    // logs the correct text based on interactions with the emp.
+
+    const spyLogger = winston.createLogger({
+      level: "info",
+      transports: [new SpyTransport({ level: "info" }, { spy: spy })]
+    });
+
+    emp = await ExpiringMultiParty.new(constructorParams);
+    eventClient = new ExpiringMultiPartyEventClient(spyLogger, ExpiringMultiParty.abi, web3, emp.address);
+    // accounts[1] is liquidator bot to monitor and accounts[2] is dispute bot to monitor.
+    contractMonitor = new ContractMonitor(spyLogger, eventClient, [accounts[1]], [accounts[2]]);
+
+    syntheticToken = await Token.at(await emp.tokenCurrency());
+
+    await collateralToken.addMember(1, tokenSponsor, {
+      from: tokenSponsor
+    });
+
+    //   Bulk mint and approve for all wallets
+    for (let i = 1; i < 5; i++) {
+      await collateralToken.mint(accounts[i], toWei("100000000"), {
+        from: tokenSponsor
+      });
+      await collateralToken.approve(emp.address, toWei("100000000"), {
+        from: accounts[i]
+      });
+      await syntheticToken.approve(emp.address, toWei("100000000"), {
+        from: accounts[i]
+      });
+    }
+
+    // Create positions for the sponsors, liquidator and disputer
+    await emp.create({ rawValue: toWei("150") }, { rawValue: toWei("50") }, { from: sponsor1 });
+    await emp.create({ rawValue: toWei("175") }, { rawValue: toWei("45") }, { from: sponsor2 });
+    await emp.create({ rawValue: toWei("1500") }, { rawValue: toWei("400") }, { from: liquidator });
+  });
+
+  it("Winston correctly emits liquidation message", async function() {
+    // Create liquidation to liquidate sponsor2 from sponsor1
+    const txObject1 = await emp.createLiquidation(
+      sponsor1,
+      { rawValue: toWei("99999") },
+      { rawValue: toWei("100") },
+      { from: liquidator }
+    );
+
+    // Update the eventClient and check it has the liquidation event stored correctly
+    await eventClient._update();
+
+    // Check for liquidation events
+    await contractMonitor.checkForNewLiquidations(() => toWei("1"));
+
+    // Ensure that the spy correctly captured the liquidation events key information.
```

Comment:
These tests are made to be independent of the structure of the output string. We can choose to re-factor the styling or ordering and these tests should still pass. Rather, these tests look for key information that the logging *must* include like links to liquidators and transactions.